### DISCUSSION TITLE
Issue with Livewire Events in Laravel 11 and Livewire 3

### DISCUSSION THREAD

**Discussion by rochap-r** [livewire/livewire#8372] [1. Help]
However, I am encountering an issue with events triggered by Livewire. When I trigger an event in my Livewire component, it seems that the event is not captured by the JavaScript in the browser.
`
    @livewireScripts
    
    <script src="{{ asset('administration/ijaboCropTool/jquery-3.6.3.min.js') }}"></script>
    <script src="{{ asset('administration/ijaboCropTool/ijabo.min.js') }}"></script>
    <script src="{{ asset('administration/ijaboCropTool/ijaboCropTool.min.js') }}"></script>
    <script src="{{ asset('administration/ijaboViewer/jquery.ijaboViewer.min.js') }}"></script>

    
    <script src="{{ asset('administration/dist/libs/apexcharts/dist/apexcharts.min.js?1674944402') }}" defer></script>
    <script src="{{ asset('administration/dist/libs/jsvectormap/dist/js/jsvectormap.min.js?1674944402') }}" defer></script>
    <script src="{{ asset('administration/dist/libs/jsvectormap/dist/maps/world.js?1674944402') }}" defer></script>
    <script src="{{ asset('administration/dist/libs/jsvectormap/dist/maps/world-merc.js?1674944402') }}" defer></script>
    
    <script src="{{ asset('administration/dist/js/tabler.min.js?1674944402') }}" defer></script>
    <script src="{{ asset('administration/dist/js/demo.min.js?1674944402') }}" defer></script>

    
    @stack('script')
    
    
    <script>
        /*window.addEventListener('showToastr', function(event) {
                            //toastr.remove()
                            if (event.detail.type === 'info') {
                                toastr.info(event.detail.message);
                            } else if (event.detail.type === 'success') {
                                toastr.success(event.detail.message);
                            } else if (event.detail.type === 'error') {
                                toastr.error(event.detail.message);
                            } else if (event.detail.type === 'warning') {
                                toastr.warning(event.detail.message);
                            } else {
                                return false;
                            }

                        });*/

        document.addEventListener('DOMContentLoaded', function() {
            window.livewire.on('showToastr', event => {
                if (event.detail.type === 'info') {
                    toastr.info(event.detail.message);
                } else if (event.detail.type === 'success') {
                    toastr.success(event.detail.message);
                } else if (event.detail.type === 'error') {
                    toastr.error(event.detail.message);
                } else if (event.detail.type === 'warning') {
                    toastr.warning(event.detail.message);
                } else {
                    return false;
                }
            });
        });

        document.getElementById("year").innerHTML = new Date().getFullYear();
    </script>`
Here is an example of code where I encounter this issue:
`<?php

namespace App\Livewire\Admin\Ui\Users;

use App\Models\User;
use Livewire\Component;
use Illuminate\Support\Facades\Hash;

class ProfilePassword extends Component
{

    public $passwordA, $passwordC,$password;

    public function changePassword(){
        
        $this->validate([
            'passwordA'=>[
                'required',function($attribute,$value,$fail){
                    if(!Hash::check($value, User::find(auth()->id())->password)){
                        return $fail(__('Le mot de que vous avez saisi est incorrect'));
                    }
                },
            ],
            'password'=>[
                'required',function($attribute,$value,$fail){
                    if(Hash::check($value, User::find(auth()->id())->password)){
                        return $fail(__('Le mot de que vous avez saisi est déjà prit!'));
                    }
                },
                'min:8','max:25',
            ],
            'passwordC'=>'same:password'
        ],[
            'passwordA.required'=>'Veuillez saisir l\'ancien mot de passe',
            'password.required'=>'Saississez le nouveau mot de passe',
            'password.min'=>'Le mot de passe doit comporter au moins 8 caractères.',
            'password.max'=>'Le mot de passe doit comporter au plus 25 caractères.',
            'passwordC.same'=>'ce mot de passe doit correspondre au nouveau mot de passe',
        ]);

        
        //chgmt du mdp
       /* $query=User::find(auth()->id())->update([
            'password'=>Hash::make($this->password)
        ]);*/
        $query=1;

        if($query){
            $this->showToastr('Votre mot de passe a été changé avec succès.','success');
            //vider les champs
            $this->passwordA=$this->password=$this->passwordC=null;
        }else{
            $this->showToastr('Quelque chose ne va pas bien.','error');
        }

    }
    public function showToastr($message,$type){
        $this->dispatch('showToastr', [
            'type' => $type,
            'message' => $message
        ]);
    }
    

    public function render()
    {
        return view('livewire.admin.ui.users.profile-password');
    }
}
`

When I run this code, my toastr message does not display in the browser. However, if I trigger the event directly in the JavaScript, the message displays correctly and the Livewire components are updated as expected.
`            $('#changeProfile').ijaboCropTool({
                preview: '',
                setRatio: 1,
                allowedExtensions: ['jpg', 'jpeg', 'png'],
                buttonsText: ['CROP', 'QUIT'],
                buttonsColor: ['#30bf7d', '#ee5155', -15],
                processUrl: '{{ route('admin.users.changePicture') }}',
                withCSRF: ['_token', '{{ csrf_token() }}'],
                onSuccess: function(message, element, status) {
                    //alert(message);
                    //listeners en ecoute pour l'actualisation de deux endroits ou se trouve les deux pictures
                    Livewire.dispatch('UpdateHeader');
                    Livewire.dispatch('UpdateProfile');
                    toastr.success(message)
                },
                onError: function(message, element, status) {
                    //alert(message);
                    toastr.error(message)
                }
            });`

Has anyone encountered this issue or have suggestions on how to resolve it? Any help would be greatly appreciated.

Thank you in advance for your time and assistance.

Best regards, Rodrigue CHOT

**Comment by lrljoe** [comment]
Please share your full layout file, wrapped in three ` marks

It is very likely that you've put the code in the wrong place in your layout.

A good package that does this out of the box is

https://github.com/masmerise/livewire-toaster
